### DISCUSSION TITLE
GoogleGroup Re: Determining the cause of a numerical instability

### DISCUSSION THREAD

**Discussion by bjoernbingham** [firemodels/fds#12670] [Q&A]
This is referring to a previous post I had on the old Google Groups page from last October: https://groups.google.com/g/fds-smv/c/DJ1VNEqtLPo/m/xSXBhKpoBQAJ

I had to substantially increase the mass flux for both the particulate and gases as well as modify the net heat flux at the PTFE boundary. Now the microgravity case (GVEC = (0,0,0)) is failing again. This is with the changes made in the previous discussion. I have tried with both DC = 0.4 or 0.04 and they fail at ~519/600 seconds and ~397/600 seconds respectively. Are there any other suggestions people have for this case? It is likely failing due to the tiny particulate causing numerical instability. Attached is a version of the case with DC = 0.04.

[fine_MicroG_HFlux_mpi_2438_cMFlux.zip](https://github.com/firemodels/fds/files/14726427/fine_MicroG_HFlux_mpi_2438_cMFlux.zip)

**Comment by mcgratta** [comment]
I will run the case with the version of FDS released yesterday.

**Comment by mcgratta** [comment]
I have these errors with the latest FDS release
```
ERROR(820): VENT Vent_out2 has no solid backing or an orientation index (IOR) is needed. (MPI Process: 15, CHID: fine_microG)
ERROR(820): VENT Vent_out2.001 has no solid backing or an orientation index (IOR) is needed. (MPI Process:  9, CHID: fine_microG)
ERROR(820): VENT Vent_in1 has no solid backing or an orientation index (IOR) is needed. (MPI Process: 10, CHID: fine_microG)
ERROR(820): VENT Vent_in2 has no solid backing or an orientation index (IOR) is needed. (MPI Process: 16, CHID: fine_microG)
```
This typically means that FDS cannot determine where to attach the VENT. Setting `THICKEN_OBSTRUCTIONS` is one possible cause.

**Comment by bjoernbingham** [comment]
Interesting, I have never gotten these errors on my locally compiled version. My version info is as follows. I you believe this could cause a numerical instability I can try changing this.

 

Fire Dynamics Simulator

 

Current Date     : March 22, 2024  10:07:44

Revision         : FDS-6.8.0-406-gc36eed0-master

Revision Date    : Thu Jul 20 04:43:12 2023 -0400

Compiler         : ifort version 2021.8.0

Compilation Date : Jul 21, 2023 12:22:55

**Comment by mcgratta** [comment]
We recently made a change to allow VENTs to be attached to thin obstructions. I wonder if this is somehow related. Can you identify these problem VENTs and see if they are unambiguously associated with a solid OBST.

**Reply by bjoernbingham** [reply]
In my blender model, they a placed on the surface of an obstruction that is not thin. I just read the updated FDS User Manual on this section and I can identify IORs if needed. I have been avoiding updating my FDS version to have continuity in all my models as I am running short on time to complete these runs now and want to avoid rerunning other gravity scenarios, but if you think the models are still comparable, then I can proceed with updating my local build.

**Comment by marcosvanella** [comment]
Do a repo update and recompile your code. You have a version from July last year of the source code.
________________________________

**Comment by mcgratta** [comment]
There is no harm in setting IOR on the VENT line. I suspect that FDS cannot determine whether the four VENTs that I cited are pointing up (+3) or down (-3). Can you tell me which?

**Comment by bjoernbingham** [comment]
That's good to know, thanks.
Vent_in1: IOR = -3
Vent_in2: IOR = -3
Vent_out2: IOR = 3
Vent_out2.001: IOR = 3

**Comment by mcgratta** [comment]
Darn, I made them all +3 and it was running. I'll run again with these values. I am using the latest release.

**Comment by bjoernbingham** [comment]
Oh shoot, the in vents are facing vertically down and the out vents are facing vertically up.

**Comment by mcgratta** [comment]
This is what I am running
```
&VENT ID='PressureEQ' SURF_ID='OPEN' XB=-0.250,0.250,-0.100,0.100,0.005,0.005 /
&VENT ID='Vent_in1' SURF_ID='FAN2' XB=-0.250,0.250,-1.000,-0.800,1.750,1.750, IOR=-3 /
&VENT ID='Vent_in2' SURF_ID='FAN2' XB=-0.250,0.250,0.800,1.000,1.750,1.750, IOR=-3 /
&VENT ID='Vent_out2' SURF_ID='FAN1' XB=-0.250,0.250,0.800,1.000,0.025,0.025, IOR=3 /
&VENT ID='Vent_out2.001' SURF_ID='FAN1'
      XB=-0.250,0.250,-1.000,-0.800,0.025,0.025, IOR=3 /
```

**Comment by mcgratta** [comment]
The job finished successfully with FDS 6.9.0.

**Comment by bjoernbingham** [comment]
Ok thanks, I will proceed with  Repo update and recompile. I will update this thread if my job runs successfully

**Comment by bjoernbingham** [comment]
Thank you, updating FDS to 6.9.0 has solved this. It ran to completion.

**Comment by mcgratta** [comment]
Great. Make sure the results look OK.

**Comment by bjoernbingham** [comment]
I just got the files downloaded. Unfortunately, there is now no flow through the ventilation. attached is a screen grab of the smokeview with a slice of the w-velocity. the flow is supposed to come in the blue vents on top and out the red vents on the floor. Did I misunderstand the IOR? 
![image](https://github.com/firemodels/fds/assets/164546233/18f43f9e-3b52-4c77-b67e-dbe840cf9918)

**Comment by mcgratta** [comment]
If the VENT is facing upward, IOR=3. If it faces downward IOR=-3. Facing up means that the gas coming out of the vent moves upwards.

**Comment by bjoernbingham** [comment]
Therefore a vent with IOR = -3 and VOLUME_FLOW = -1 would blow down, correct?

**Comment by mcgratta** [comment]
Yes. IOR=-3 means that the vent is facing downward, and VF=-1 means that the vent is blowing outwards into the domain, as opposed to sucking inwards.

**Comment by bjoernbingham** [comment]
Ok, so that is not the issue. I have been messing with a few settings now with no luck yet. I refined the positions of the vents to ensure they are correctly located relative to the walls, turning off thicken obstructions, and trying a different mesh. Are there any reasons you can think of that would cause the vents to have no flow?

**Comment by rmcdermo** [comment]
@bjoernbingham I'm a little lost at this point as to what input file we should be testing.  Could you please simplify the case as much as possible and post an input file that that shows not flow through the vent.  Thanks

Also, please unmark the discussion as "answered" if we still have work to do.  Thanks

**Comment by bjoernbingham** [comment]
It appears the mesh/vent alignment was off. I successfully was able to get a 10-second test case to initialize with vents flowing.
